detection_rules/schemas/registry_package.py

```diff
@@ -0,0 +1,53 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+
+"""Definitions for packages destined for the registry."""
+
+import dataclasses
+from typing import Dict, Union, Type
+
+import marshmallow_dataclass
+from marshmallow import validate
+
+from .definitions import BaseMarshmallowDataclass, ConditionSemVer, SemVer
+
+
+@marshmallow_dataclass.dataclass
+class BaseManifest(BaseMarshmallowDataclass):
+    """Base class for registry packages."""
+
+    conditions: Dict[str, ConditionSemVer]
+    version: SemVer
+    format_version: SemVer
+
+    categories: list = dataclasses.field(default_factory=lambda: ['security'].copy())
+    description: str = 'Rules for the detection engine in the Security application.'
+    icons: list = dataclasses.field(default_factory=list)
+    license: str = 'basic'
+    name: str = 'detection_rules'
+    owner: dict = dataclasses.field(default_factory=lambda: dict(github='elastic/protections').copy())
+    policy_templates: list = dataclasses.field(default_factory=list)
+    release: str = 'experimental'
+    screenshots: list = dataclasses.field(default_factory=list)
+    title: str = 'Detection rules'
+    type: str = 'rules'
+
+
+@marshmallow_dataclass.dataclass
+class ManifestV1Dot0(BaseManifest):
```

Comment:
i'm thinking that for now we start small and don't have multiple manifest versions.
i'm not sure what future manifests will look like and how they'll differ so i can't make advice on how to structure this  process

Reply:
Hopefully they remain relatively stable. I can table this approach and go off a single manifest assumption then and if needed, re-introduce some of this code